feature S3C-4213: Add docs for using vault (#3481)
scality/cloudserver: README.md

a secret key of verySecretKey1.

## Run it with Vault user management

> Note: Vault is a private product and you will need access to it separately.

Reviewer: ```suggestion
Note: Vault is proprietary and must be accessed separately.
```

Author: I spent like 5 minutes thinking about how to best describe this, before I realized I had no idea what terms to use and just went with whatever... thanks

Reviewer: No worries, matey. Words are what I do!
